test/framework/backuppolicy/backuppolicy.go

```diff
@@ -0,0 +1,182 @@
+	p := &v1beta3.BackupPolicy{}
+	for _, opt := range opts {
+		opt(p)
+	}
+	return p
+}
+
+func GetExistingBackups(p *v1beta3.BackupPolicy, dsi runtimeClient.Object) []v1beta3.Backup {
+	existingBackups := []v1beta3.Backup{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              framework.UniqueName("backup", suffixLength),
+				Namespace:         p.Namespace,
+				CreationTimestamp: v1.Time{Time: mustParseTime("2024-07-03T00:00:00Z")},
+				Labels: map[string]string{
+					v1beta3.BackupPolicyNameLabelKey: p.Name,
+				},
+			},
+			Spec: v1beta3.BackupSpec{
+				ServiceInstance: p.Spec.ServiceInstance,
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              framework.UniqueName("backup", suffixLength),
+				Namespace:         p.Namespace,
+				CreationTimestamp: v1.Time{Time: mustParseTime("2024-07-02T00:00:00Z")},
+				Labels: map[string]string{
+					v1beta3.BackupPolicyNameLabelKey: p.Name,
+				},
+			},
+			Spec: v1beta3.BackupSpec{
+				ServiceInstance: p.Spec.ServiceInstance,
+			},
+		},
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:              framework.UniqueName("backup", suffixLength),
+				Namespace:         p.Namespace,
+				CreationTimestamp: v1.Time{Time: mustParseTime("2024-07-01T00:00:00Z")},
+				Labels: map[string]string{
+					v1beta3.BackupPolicyNameLabelKey: p.Name,
+				},
+			},
+			Spec: v1beta3.BackupSpec{
+				ServiceInstance: p.Spec.ServiceInstance,
+			},
+		},
+	}
+	return existingBackups
+}
+
+// CheckBackups waits for the backups matching length.
+func CheckBackups(ctx context.Context, p *v1beta3.BackupPolicy, c runtimeClient.Client, expectedLength int) *v1beta3.BackupList {
+	var err error
+	backups := &v1beta3.BackupList{}
+	EventuallyWithOffset(1, func() bool {
+		lisOptions := []client.ListOption{
+			client.InNamespace(p.Namespace),
+			client.MatchingLabels{v1beta3.BackupPolicyNameLabelKey: p.Name},
+		}
+
+		if err = c.List(ctx, backups, lisOptions...); err != nil {
+			return false
+		}
+
+		if len(backups.Items) == expectedLength {
+			return true
+		}
+		return false
+	}, asyncOpsTimeoutMins, pollingPeriod).Should(BeTrue(),
+		fmt.Sprintf("timeout reached waiting for correct backups count %s/%s: %s",
+			p.GetNamespace(),
+			p.GetName(),
+			err,
+		),
+	)
+	return backups
+}
+
+func CreateBackup(ctx context.Context, k8sClient runtimeClient.Client, namespace, creationTime string) {
+	backup := &v1beta3.Backup{
+		ObjectMeta: v1.ObjectMeta{
+			Name:              framework.UniqueName("backup", suffixLength),
+			Namespace:         namespace,
+			CreationTimestamp: v1.Time{Time: mustParseTime(creationTime)},
+		},
+	}
+	k8sClient.Create(ctx, backup)
+}
+
+func mustParseTime(timeStr string) time.Time {
+	parsedTime, err := time.Parse(time.RFC3339, timeStr)
+	if err != nil {
+		panic(fmt.Errorf("error parsing time %w", err))
+	}
+	return parsedTime
+}
+
+func CleanupBackups(ctx context.Context, p *v1beta3.BackupPolicy, c runtimeClient.Client) error {
```

Comment:
Question: shouldn't the backup policy clean up the backups? Why do we need a utility function in the test library? Do we test that all backups are cleaned up when the policy is deleted?

Reply:
@lhaendler backups will be deleted if retentionConfig is defined, we need it to clean the remaining backups also.
Now if the policy is deleted, no cascading deletion to the backups

Reply:
@rfashwall-anynines and I have discussed this and we have come to the conclusion that this could be configurable, a ticket has been created to handle it in the future